### ISSUE TITLE
AL-Validation fails if Next Major is not available

### ISSUE THREAD

**Issue by makuhn** [microsoft/navcontainerhelper#2381]
**Describe the issue**
We started our Al-Validation pipeline this morning 8:50 which runs the validation for 13 apps and 11 countries per app. Pipeline is composed of multiple stages (one stage per app) which are processed by two agents on same virtual machine. Each stage basically runs with same parameters (see extract below).

```
 |  __ \                             | |               
 | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
 |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
 | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
 |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/

Container name               ALVal-LotMgt
Credential                   admin/Xotu2005
MemoryLimit                  8G
validateVersion              
validateCurrent              True
validateNextMinor            False
validateNextMajor            True
SasToken                     Specified
countries                    de,nl,us,be,fr,at,ch,se,au,no,it
validateCountries            de,nl,us,be,fr,at,ch,se,au,no,it
affixes                      PB365
supportedCountries           
vsixFile                     
```

After a while all stages did crash with following error:
```
Removing container ALVal-LotMgt
Removing ALVal-LotMgt from container hosts file
Removing ALVal-LotMgt-* from container hosts file
Removing C:\ProgramData\BcContainerHelper\Extensions\ALVal-LotMgt

Removing container took 4 seconds
  _____       _                      _                         _   _  __           _       
 |  __ \     | |                    (_)                       | | (_)/ _|         | |      
 | |  | | ___| |_ ___ _ __ _ __ ___  _ _ __   ___    __ _ _ __| |_ _| |_ __ _  ___| |_ ___ 
 | |  | |/ _ \ __/ _ \ '__| '_ ` _ \| | '_ \ / _ \  / _` | '__| __| |  _/ _` |/ __| __/ __|
 | |__| |  __/ |_  __/ |  | | | | | | | | | |  __/ | (_| | |  | |_| | || (_| | (__| |_\__ \
 |_____/ \___|\__\___|_|  |_| |_| |_|_|_| |_|\___|  \__,_|_|   \__|_|_| \__,_|\___|\__|___/
                                                                                           
##[error]ForEach-Object : You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.2\AppHandling\Run-AlValidation.ps1:140 char:22
+         $countries | ForEach-Object {
+                      ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [ForEach-Object], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull,Microsoft.PowerShell.Commands.ForEachObjectCommand
```

Having a more detailed look into the logs we recognized that current artifacts version did change from 

log extract of successfully validated app/stage:
```
2022-04-01T08:19:20.6857465Z  |  __ \                             | |               
2022-04-01T08:19:20.6864040Z  | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
2022-04-01T08:19:20.6873230Z  |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
2022-04-01T08:19:20.6877506Z  | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
2022-04-01T08:19:20.6886990Z  |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/
2022-04-01T08:19:20.6887523Z 
2022-04-01T08:19:20.6892116Z Container name               ALVal-Compl
2022-04-01T08:19:20.7008023Z Credential                   admin/Zipi2639
2022-04-01T08:19:20.7637992Z MemoryLimit                  8G
2022-04-01T08:19:20.7644947Z validateVersion              
2022-04-01T08:19:20.7656202Z validateCurrent              True
2022-04-01T08:19:20.7666871Z validateNextMinor            False
2022-04-01T08:19:20.7677718Z validateNextMajor            True
2022-04-01T08:19:20.7688686Z SasToken                     Specified
2022-04-01T08:19:20.7730023Z countries                    de,nl,us,be,fr,at,ch,se,au,no,it
2022-04-01T08:19:20.7741090Z validateCountries            de,nl,us,be,fr,at,ch,se,au,no,it
2022-04-01T08:19:20.7765133Z affixes                      PB365
2022-04-01T08:19:20.7788062Z supportedCountries           
2022-04-01T08:19:20.7797364Z vsixFile                     
2022-04-01T08:19:20.7802837Z Install Apps
2022-04-01T08:19:20.7808247Z - None
2022-04-01T08:19:20.7813903Z Previous Apps
2022-04-01T08:19:20.7838279Z - E:\agents\a1\_work\6\s\tools\Temp\Previous\apps\ProBatch4.2.73.5\apps\YAVEON AG_ProBatch 365 Base_4.2.73.5.app
2022-04-01T08:19:20.7844148Z - E:\agents\a1\_work\6\s\tools\Temp\Previous\apps\ProBatch4.2.73.5\apps\YAVEON AG_ProBatch 365 Compliance_4.2.73.5.app
2022-04-01T08:19:20.7849724Z Apps
2022-04-01T08:19:20.7872150Z - E:\agents\a1\_work\6\s\tools\Temp\Current\ProBatch4.3.84.1\apps\YAVEON AG_ProBatch 365 Base_4.3.84.1.app
2022-04-01T08:19:20.7877715Z - E:\agents\a1\_work\6\s\tools\Temp\Current\ProBatch4.3.84.1\apps\YAVEON AG_ProBatch 365 Compliance_4.3.84.1.app
2022-04-01T08:19:20.7934796Z 
2022-04-01T08:19:20.7938770Z   _____       _ _ _                                          _        _                            
2022-04-01T08:19:20.7946357Z  |  __ \     | | (_)                                        (_)      (_)                           
2022-04-01T08:19:20.7950230Z  | |__) |   _| | |_ _ __   __ _    __ _  ___ _ __   ___ _ __ _  ___   _ _ __ ___   __ _  __ _  ___ 
2022-04-01T08:19:20.7955084Z  |  ___/ | | | | | | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '__| |/ __| | | '_ ` _ \ / _` |/ _` |/ _ \
2022-04-01T08:19:20.7961692Z  | |   | |_| | | | | | | | (_| | | (_| |  __/ | | |  __/ |  | | (__  | | | | | | | (_| | (_| |  __/
2022-04-01T08:19:20.7970735Z  |_|    \__,_|_|_|_|_| |_|\__, |  \__, |\___|_| |_|\___|_|  |_|\___| |_|_| |_| |_|\__,_|\__, |\___|
2022-04-01T08:19:20.7975565Z                            __/ |   __/ |                                                 __/ |     
2022-04-01T08:19:20.7984900Z                           |___/   |___/                                                 |___/      
2022-04-01T08:19:20.7986760Z 
2022-04-01T08:19:20.7991681Z Pulling mcr.microsoft.com/businesscentral:10.0.17763.2686-filesonly
2022-04-01T08:19:21.5058232Z 
2022-04-01T08:19:21.5067443Z Pulling generic image took 1 seconds
2022-04-01T08:19:21.5408459Z   _____       _                      _                         _   _  __           _       
2022-04-01T08:19:21.5416822Z  |  __ \     | |                    (_)                       | | (_)/ _|         | |      
2022-04-01T08:19:21.5421885Z  | |  | | ___| |_ ___ _ __ _ __ ___  _ _ __   ___    __ _ _ __| |_ _| |_ __ _  ___| |_ ___ 
2022-04-01T08:19:21.5427899Z  | |  | |/ _ \ __/ _ \ '__| '_ ` _ \| | '_ \ / _ \  / _` | '__| __| |  _/ _` |/ __| __/ __|
2022-04-01T08:19:21.5434413Z  | |__| |  __/ |_  __/ |  | | | | | | | | | |  __/ | (_| | |  | |_| | || (_| | (__| |_\__ \
2022-04-01T08:19:21.5442707Z  |_____/ \___|\__\___|_|  |_| |_| |_|_|_| |_|\___|  \__,_|_|   \__|_|_| \__,_|\___|\__|___/
2022-04-01T08:19:21.5446193Z                                                                                            
2022-04-01T08:19:30.4398519Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/de
2022-04-01T08:19:38.9862457Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/nl
2022-04-01T08:19:47.4310406Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/us
2022-04-01T08:19:56.2617212Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/be
2022-04-01T08:20:04.9297786Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/fr
2022-04-01T08:20:13.8961587Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/at
2022-04-01T08:20:22.5991064Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/ch
2022-04-01T08:20:31.2941861Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/se
2022-04-01T08:20:39.9748202Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/au
2022-04-01T08:20:48.7179174Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/no
2022-04-01T08:20:57.5615385Z Found https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/it
2022-04-01T08:20:57.6157024Z Using https://bcartifacts.azureedge.net/sandbox/19.5.36567.38429/de
```

Log extract of failing app/stage:
```
2022-04-01T09:03:11.5096568Z  |  __ \                             | |               
2022-04-01T09:03:11.5102926Z  | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
2022-04-01T09:03:11.5109779Z  |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
2022-04-01T09:03:11.5117730Z  | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
2022-04-01T09:03:11.5124204Z  |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/
2022-04-01T09:03:11.5127998Z 
2022-04-01T09:03:11.5135507Z Container name               ALVal-LotMgt
2022-04-01T09:03:11.5409488Z Credential                   admin/Xotu2005
2022-04-01T09:03:11.5886128Z MemoryLimit                  8G
2022-04-01T09:03:11.5894834Z validateVersion              
2022-04-01T09:03:11.5911675Z validateCurrent              True
2022-04-01T09:03:11.5922417Z validateNextMinor            False
2022-04-01T09:03:11.5934566Z validateNextMajor            True
2022-04-01T09:03:11.5950332Z SasToken                     Specified
2022-04-01T09:03:11.5983278Z countries                    de,nl,us,be,fr,at,ch,se,au,no,it
2022-04-01T09:03:11.5996344Z validateCountries            de,nl,us,be,fr,at,ch,se,au,no,it
2022-04-01T09:03:11.6021542Z affixes                      PB365
2022-04-01T09:03:11.6045805Z supportedCountries           
2022-04-01T09:03:11.6055703Z vsixFile                     
2022-04-01T09:03:11.6062440Z Install Apps
2022-04-01T09:03:11.6068392Z - None
2022-04-01T09:03:11.6074567Z Previous Apps
2022-04-01T09:03:11.6098782Z - E:\agents\a1\_work\6\s\tools\Temp\Previous\apps\ProBatch4.2.73.5\apps\YAVEON AG_ProBatch 365 Base_4.2.73.5.app
2022-04-01T09:03:11.6105358Z - E:\agents\a1\_work\6\s\tools\Temp\Previous\apps\ProBatch4.2.73.5\apps\YAVEON AG_ProBatch 365 Lot Management_4.2.73.5.app
2022-04-01T09:03:11.6110466Z Apps
2022-04-01T09:03:11.6133193Z - E:\agents\a1\_work\6\s\tools\Temp\Current\ProBatch4.3.84.1\apps\YAVEON AG_ProBatch 365 Base_4.3.84.1.app
2022-04-01T09:03:11.6139781Z - E:\agents\a1\_work\6\s\tools\Temp\Current\ProBatch4.3.84.1\apps\YAVEON AG_ProBatch 365 Lot Management_4.3.84.1.app
2022-04-01T09:03:11.6200224Z 
2022-04-01T09:03:11.6207031Z   _____       _ _ _                                          _        _                            
2022-04-01T09:03:11.6212895Z  |  __ \     | | (_)                                        (_)      (_)                           
2022-04-01T09:03:11.6218740Z  | |__) |   _| | |_ _ __   __ _    __ _  ___ _ __   ___ _ __ _  ___   _ _ __ ___   __ _  __ _  ___ 
2022-04-01T09:03:11.6224874Z  |  ___/ | | | | | | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '__| |/ __| | | '_ ` _ \ / _` |/ _` |/ _ \
2022-04-01T09:03:11.6230348Z  | |   | |_| | | | | | | | (_| | | (_| |  __/ | | |  __/ |  | | (__  | | | | | | | (_| | (_| |  __/
2022-04-01T09:03:11.6235615Z  |_|    \__,_|_|_|_|_| |_|\__, |  \__, |\___|_| |_|\___|_|  |_|\___| |_|_| |_| |_|\__,_|\__, |\___|
2022-04-01T09:03:11.6241845Z                            __/ |   __/ |                                                 __/ |     
2022-04-01T09:03:11.6252087Z                           |___/   |___/                                                 |___/      
2022-04-01T09:03:11.6255301Z 
2022-04-01T09:03:11.6261560Z Pulling mcr.microsoft.com/businesscentral:10.0.17763.2686-filesonly
2022-04-01T09:03:12.3458926Z 
2022-04-01T09:03:12.3465328Z Pulling generic image took 1 seconds
2022-04-01T09:03:12.3751600Z   _____       _                      _                         _   _  __           _       
2022-04-01T09:03:12.3758062Z  |  __ \     | |                    (_)                       | | (_)/ _|         | |      
2022-04-01T09:03:12.3764734Z  | |  | | ___| |_ ___ _ __ _ __ ___  _ _ __   ___    __ _ _ __| |_ _| |_ __ _  ___| |_ ___ 
2022-04-01T09:03:12.3771421Z  | |  | |/ _ \ __/ _ \ '__| '_ ` _ \| | '_ \ / _ \  / _` | '__| __| |  _/ _` |/ __| __/ __|
2022-04-01T09:03:12.3778408Z  | |__| |  __/ |_  __/ |  | | | | | | | | | |  __/ | (_| | |  | |_| | || (_| | (__| |_\__ \
2022-04-01T09:03:12.3784453Z  |_____/ \___|\__\___|_|  |_| |_| |_|_|_| |_|\___|  \__,_|_|   \__|_|_| \__,_|\___|\__|___/
2022-04-01T09:03:12.3790752Z                                                                                            
2022-04-01T09:03:21.0426391Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/de
2022-04-01T09:03:29.6110188Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/nl
2022-04-01T09:03:38.4116845Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/us
2022-04-01T09:03:46.6779579Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/be
2022-04-01T09:03:55.0232123Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/fr
2022-04-01T09:04:03.2175586Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/at
2022-04-01T09:04:13.3442826Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/ch
2022-04-01T09:04:21.3854904Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/se
2022-04-01T09:04:29.5593534Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/au
2022-04-01T09:04:37.7773801Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/no
2022-04-01T09:04:45.9759311Z Found https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/it
2022-04-01T09:04:46.0276949Z Using https://bcartifacts.azureedge.net/sandbox/20.0.37253.38362/de
```

It seems that BC artifacts where updated during the pipeline execution and next major version was moved to current.

I can share the full log if necessary.

**Comment by freddydk** [comment]
I will make the code more resilient.
Right after release there are some hours or days where next major is unavailable - it shouldn't crash, it should just validate what is and write a notification.

**Comment by freddydk** [comment]
Shipped in 3.0.5
